Deepfrying : MAKE SPACE STATION 13 GREAT AGAIN

Iamgoofball opened the pull request:
# dict. trim occured on meta
![dreamseeker_2017-02-01_18-23-39](https://cloud.githubusercontent.com/assets/4081722/22534842/dceb0b3c-e8ab-11e6-9c60-565bc79fd406.png)
# Q&A
![vlcsnap-2012-09-22-17h21m01s203](https://cloud.githubusercontent.com/assets/4081722/22623616/f2fd844c-eb15-11e6-99f0-aa7d0be7b3de.png)

Iamgoofball committed:
MAKE SPACE STATION 13 GREAT AGAIN

Cyberboss commented:
A goof PR that's not trash???

ma44 commented:
Questions

1. When can i deep fry lizards?

2. When can i impose a ban on those damm lizards immigrating inside of our station?

3. When can i take the container with the oil and throw it on a person?

4. Can i get heart attacks from this?

5. Will this contain bugs? (why am i even asking it's goofcode)

6. Will this allow me to deepfry lemons and dosh bills?

GunHog commented:
The deep fried deep fried deep fried deep fried deep fried deep fried deep fried deep fried nuclear authentication disk

Iamgoofball commented:
1. YES
2. YES
3. YOU CAN DUNK PEOPLE
4. YES
5. MORE LIKE EDGE CASES
6. YES

ma44 commented:
Can i use facid as an alternative to oil/cornoil for muh INTER DEPARTMENT COOPERATION

Iamgoofball commented:
1. yes

bgobandit commented:
meme PR, no one liked this then and no one should like it now

Iamgoofball commented:
1. unamerican spotted, SAD!

anconfuzedrock commented:
is it possible to move this with wrench/make more/spill it/wire one directly to disposals

GunHog commented:
Please let me be able to print circuits so I can build 5 more of these for the chef!

Iamgoofball committed:
MAP CHANGES

Iamgoofball commented:
@anconfuzedrock 
![house_306-queserasera_1163025302](https://cloud.githubusercontent.com/assets/4081722/22535398/c6d784de-e8af-11e6-886b-63f2f165b15a.jpg)

PKPenguin321 commented:
This is one of my favorite features of all time. Please let it pass this time.

Iamgoofball commented:
@PKPenguin321 
![chrome_2017-02-01_19-00-38](https://cloud.githubusercontent.com/assets/4081722/22535555/bf7696e8-e8b0-11e6-9389-3cd245bf4cc5.png)

Kromgar commented:
... wasn't this a thing a long time ago?

Oh that was that uh code base split i think... didn't last long i think a month or something?

Iamgoofball commented:
@Kromgar 
![chrome_2017-02-01_19-37-27](https://cloud.githubusercontent.com/assets/4081722/22536327/e109f41c-e8b5-11e6-90a9-762468f66db0.png)

PKPenguin321 commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,315 @@
+
+/obj/machinery/deepfryer/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/deepfryholder))
+		user << "<span class='userdanger'>Your cooking skills are not up to the legendary Doublefry technique.</span>"
```
Lame as fuck

kevinz000 replied:
ikr

Iamgoofball replied:
i'll add a cookbook that fixes this

ma44 replied:
Should check if the user has a chef job, as chefs are the master cooks

Supermichael777 replied:
@ma44 it needs to be a var in mind and you can set it when you equip the chefs  as well as with other objects

PKPenguin321 commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,315 @@
+			return
+	else if(user.pulling && user.a_intent == "grab" && isliving(user.pulling) && reagents.total_volume)
+		var/mob/living/L = user.pulling
+		if(!iscarbon(L))
```
Just change the isliving check in the else if above to iscarbon

PKPenguin321 reviewed (commented):
Please let this get merged

kevinz000 commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,315 @@
+		reagents.reaction(C, TOUCH)
+		C.adjustFireLoss(reagents.total_volume)
+		reagents.remove_any((reagents.total_volume/2))
+		C.Weaken(3)
```
No.

Iamgoofball replied:
you just got dunked into a vat of boiling reagents

Fox-McCloud replied:
This will need a click cooldown.

PKPenguin321 replied:
No, but it needs to break your grab if it doesn't already. Being able to chain it (like chain tabling) is fine.

PKPenguin321 replied:
Fixed on the line below this one

Buggy123 commented:
Goof best coder confirmed.

ma44 commented:
Let's not all start going crazy over here

calzilla1 commented:
ITGOESITGOESITGOESITGOESITGOES

Haevacht commented:
So can you still deep fry a nuke disk?

jughu commented:
Can i replace the oil with blwckpowder

jughu commented:
@iamgoofball the double fry technique better be a traitor item

kevinz000 commented:
@Iamgoofball please no ascii art in the code.

ma44 reviewed (commented):
@Iamgoofball You can stun someone forever with this just by pressing a hotkey button and clicking the deep fryer 👍

Iamgoofball commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,315 @@
+/obj/machinery/deepfryer/attack_hand(mob/user)
+	if(frying)
+		if(frying.loc == src)
+			user << "<span class='notice'>You eject [frying] from [src].</span>"
```
Doesn't matter

RemieRichards commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
````diff
@@ -0,0 +1,315 @@
+April 3rd, 2014 marks the day this machine changed the face of the kitchen on NTStation13
+God bless America.
+
+'+++++'++;,...,':+##+@@#####@@@@@@@@@@@@@@@@@@@@##+'`..,,,'''''''''''+##++'',```..,:;+#######++++''++++;.....:+++##++''''''''''+'+'''+++++++#+++########################+';;:,,,,,,,::::::'++###########
````
The PR will NOT be merged with this comment in it.

Iamgoofball replied:
Communist detected

ExcessiveUseOfCobblestone replied:
See a Red? Shoot 'em Dead!

PKPenguin321 replied:
Please goof just do what they ask I need deep friers please I am begging you

Iamgoofball replied:
i don't understand the problem with this comment

PKPenguin321 replied:
Please just do what they ask goof please

Arianya commented:
I can't see anything that stops the nuke disc or other problematic items from being dunked in here.

Does a deep fried nuke disc still work as a nuke disc? If not, does it spawn a new nuke disc on deep frying?

Iamgoofball commented:
It'll need to be un-fried by being eaten which will make it respawn

Iamgoofball commented:
It used to just delete the disk on fry but people complained

Iamgoofball commented:
>>>>>>>>>>>>map conflicts

Arianya commented:
Why not have it spawn a new nuke disc on frying? Also do pinpointers still point to a fried nuke disc properly?

Also also what happens if you deep fry a bluespace body bag with someone in it?

jughu commented:
@arianya did it a round ago nothing funny happened

Iamgoofball commented:
Pinpointers work.

I dunno.

dannno commented:
is recursive frying still a thing
because it's actually not that funny!

Iamgoofball commented:
nope

Iamgoofball committed:
'+++++'++;,...,':+##+@@#####@@@@@@@@@@@@@@@@@@@@##+'`..,,,'''''''''''+##++'',```..,:;+#######++++''++++;.....:+++##++''''''''''+'+'''+++++++#+++########################+';;:,,,,,,,::::::'++###########
+'++++'++;;`.,.;'+##########@@@@@@@@@@@@@@@@@@@@@+++;...,,+'''''''''''###+'''.```.,;#+#######+#+++++++:'.,...;:++###++'''''''''+'+''+++#++++#++++########################''';:,,,,,:::::::;'##+#########
++'++'';:.:`,..:;++#@########@@@@@@@@@@@@@@@@@@@@++#'....;''''''''''''+##++'':```.::+###########'+++++::.....;;+++###+''''''''''+'''+++++++++++++#########################+;'',,,,,,::::::::'##+########
+++++++::.,`..,.+++#########@@@@@#@@@@@@@@@@@@@@@@++;:..::'''''''''''''++++'',. .`.#+##+########+##++++,..`..,+++++##'+''''''''''+'+++++++++++#+++##+######################+;'',,,,,::::::::::+#+#######
++''++';:,..`.;'+++##########@@@@@#@@@@@@@@@@@@@@@+';:;,::+''''''+''''''#++'+,:+`::+++##+##########+++++...`.'++++++##:'''''''''''+'''++++++++++++#+#+#######################''',,::,:::::::::;;+#######
++++'+++:...``;+''++##########@@@@#@@#@@@#@@@@@@@@#;:++#::+''''''''''''''#+':,;'+,,+++###+##########+++''`...#+++++++##::;;;'''''''''''+'+++++++++++++###+####################';';,,,::::::::::;::+##+##
+++++'+++`..,`;++'++##########@@@@+#@#@##@@@@@@@@@@:'++'+,+'''''''''''''''++,+'''+,++++##############++'+`.``;++++++++#;::,,``..:'++''+'++++++++++++++##+######################+;;';;;:::::::::;:::;+###
++++++++':.,;:,'+++++##########@@@@'+@+++@@@@@@@@@@#+++'+#''''''''''''''''+++'''''+++++##############+'++:`.:;+++++++++#:::...``.` `,'+'''+++++++++++++##########################;;;;;;:::::::::;;:;;;##
'+++++'++,:'':,+++++++##########@@##+#'+++#@@@@@@@@##++#'''+'''''''''''''''+#''''''++++###############+'+:;:::+++++++++++:,...`.````````:+++++++++++++++##+#######################';;;;;;;:;;::;;;;;;;:'
'++++++'#:'+'+,+++++++############@@++';+++#@@@@@@@@#++++++++'+'''''''''''''+#'''''+'#+###############+':::+;:+'+++++++'+',,...```````.....:'+++++++++++++#########################'';;;;;;;;;;;;;;;;;;;
++++++++;+''++++++++++#############@@;':++#@@@@@@@@@@#+++++++'+''''''''''''''++'''''+'#+##############'+:;+'+:'+++++++++'+:':.``````````.`..,.;++++++++++#++########################';;;;;;;;;;;;;;;;;;;
#+++#+++'+'''+'++++++++############@##'+:##+#@@@@@@@@@#++++'+++''''''''''''''+++'''''++#+#############++;''''+,++++++++++'+:+;.``````````......,:'++++++++#+##########################;;;;;;;;;;;;;;;;;;
+++++++'+'+''+++++++++++##########@@#+#+''++##@@@@@@@@##+++'++''''''''''''''''+#''''++++#+############+'+'''''+++++++++++''+,':,`````````.....,,,,,;+++++++#+##########################';;;;;;;;;;;;;;;;
++#'+#+'++''''+'++++++++##############+#+;+#@@@@@@@@@@@#+++'+'+''''''''''''''''++''''+++####@##########''''''''+++++++++++''+;+:,`````````......,,,::;+++++#++##########################';;;;;;;;;;;;;;;
++#+++#+''+''''++++++++++#+##########@@#@+'@@@@@@@@@@@@##++++++'''''''''''''''''+#''''++##+############'''''''+++++++++++++'';++;.`````````.....,,:,,,,;++++++#+#########################';;;;;;;;;;;;;;
+++#++#++'+''''++++'++++++++########@@@#@@'#@@@@@@@@@@@@#+++'+''''''''+''''''''''++''++####+##########++'+'''+''++++++++++++++'++.,````````.....,,,,,,,.,:++++#++#########################+';;:;:;;;;;;;
+++##++++'''''''''+''+++++++########@@@@#@@+@@@@@@@@@@@@##++++''''';'''+'''''''''+++'+++####+#########++++''''''+++++++++++++++++',,.``````.....,,,.,,,,,:::+++++##########################+';::::::::;;
++++#+'+#+'''''''''+''++':+#+###@##@@@@@@@@#@@@@@@@@@@@@@##+++''''',:;''+'''''''''+++'++####'##########++''''''++++++++++++++++++#.,,.`````......,,,,,..,,::,,+#+#+##########################'':::::::::
+++++#+'+#+''''''+''+'+',,++#+#####@@@@@@@@@#@@@@@@@@@@@@@#+++'''';.`:''+''''''''''++'++##+;;+#########++++'''+++++++'+++++++#++++'.,..`````......,,,,.,,,,,,,,;++#++#########################'';:::::::
++++'##''++'''''''''''':.:';#+######@@@@@@@@@@@@@@@@@@@@@@##+++''', .,,,,'''''''''''+#:::+',::+########+++++''++++++++++++++++++++#.....````......,,,.,,,,,,,,,:::++++#########################'';::::::
+++'''++'+++''''''''+''.,::;+#######@@@@@@@@@@@@@@@@@@@@@@###+++',: ``.,+'''''''''''++#,;:;:;';#########++++'+'++++++++++++++++++++:,....````.........,,,,,,,,,,::,:#++#########################'';,::::
+++'+'+++''++'''''':,;`.,,:,'+#+#####@@@@@@@@@@@@@@@@@@@@@@###+,:.`` ``;'''''''''''''++;;,.;,;;'########+++++''++++++'+++''++++++++#`.....````...,..,,,,,,,,,,,,::,::'#++########################+''';::
;;++'''+#+''+''''+'':....,,::'#######@@@@@@@@@@@@@@@@@@@@@@@##+':`````,+''''''''''''''++:..,',;,#########+++++++#++''+++++''++++++++;.``..`````....,,,,,,,,,,,,,,::,::;'#+#+######################+;'';:
,.:'''''++''''''''+';,.`..,;:+####@@#@@@@@@@@@@@@@@@@@@@@@@@###+,`````'''''''''''''''''#+.,,:'::+#########+++++#+++'+++','+++++++++++`.```.`````...,,,,,,,,,,,,,,,::,:;;:+#+########################;'';
...;'''''++'''''''++',..`..''+####@@@@@@@@@@@@@@@@@@@@@@@@@@@#+++:```.:'''''''''''''+''+;;.,:;'++##########++++++++++++;.,:+''++++++++`.````````....,,,,,,,,,,,,,,::,::;;::##########################';'
...''':''+++'''''''+':.`,,.;#+####@@@@@@@@@@@@@@@@@@@@@@@@@@@##+'+ ``:,'''''''''''''+++';+++,:;#############++++###++';'.`:'''''+++++++..```````....,,,,,,,,,,,,,,:::::;;;;,++########################';
...`,:'+''+++'''''''#::+:,:;#######@@@@@@@@@@@@@@@@@@@@@@@@@@@@++;:.,,,'''''''''''+'+++;'+++':'#############+++####;;+,:..,:++++++++++++:````````...,,,,,,,,,,,,,,::::::;:;;;:#########################+
....`:+''''+++''''''::'++;;;#######@@@@@@@@@@@@@@@@@@@@@@@@@@@@++:,;;,,''''''''''+'+++':+++++':#############++#####;:;`,,...;:+'+''++++++' ``````...,,,,,,,,,,,,,,,:::::;;;:,':+########################
...`:'''''''++'''''+:++++++:########@@@@@@@@@@@@@@@@@@@@@@@@@@@#':;+'',+''''''''''+++'+++++++#'##############+######:,...,:::;'++++'+++++++```````...,.,,,,,,,,,,:::::::::;::,;;'#######################
...`;''''''''++'''''+++++#++#########@@@@@@@@@@@#@@@@@@@@@@@@@@@;:+''''+'''''''''''+++'+++++++#+###################+#:::.```.......,:'++''++``````...,,,,,,,,,,,,:::::::::;::::::'######################
...`+'''''''''++'''''+++++#+###########@@@@@@@@@@+@@@#@@@@@@@@@@#++'+''''''''''''+'++++++'++++##+###############@#+',.`` `.````````....,'+++' `````..,,,,,,,,,,,,::::::::::;::,;:;;#####################
...:'''''''''''+'''''+++++##+############@@@@@##@###++#@@@@@@@@@@+++'+''''''''''''+'+'''+++'+++#################;.`  ``````````````````.`.;++;`..```,,,,,,,,,,,,,:::::::::::::::;;';+###################
`.::+'''''''''''''''''+++++#+############@@@@@##@@##+++@@@@@@@@@@#+++''''''''''''''+'+'''+'+++++++###########+. ` `````````````````````````.'+,`..``,,,,,,,,,,,:,,:::::::::::::,;:;;;+##################
++::''''''''''''''''''+++++++############@@@@@@###@++'++@@@@@@@@@@+++'''''''''''''''+''''''''++++##########+.  `  `.`````````````````````````:+``.``.,,,,:,,,,,:::,::::::::::::::;;;;;+#################
''+';''''''''''''''''''++++++############@@#@@@#####++###@@@@@@@@@#++'''''''''''''''''''''''''++++#######'.   `  ``.````````` ```  `````.```` `'...`.,,,,::,,,,::::::::::::::::;:;;;;;;'################
'''''''''''''''''''''''++++++############@@@@@@#####++####@@@@@@@@##+''''''''''''''''''''''''''++++####' `` ``  ````` ` ```````````````````.,:.``,,..,:,,,::,,,::::::::::::::::;::;;;;';'###############
'''''''''''''''''''''''+++++++###########@@@#@@@##########@@@@@@@@@#+'''''''''''''''''+'''''''''++++##:     `  ```` `` ``````````````````````.,,:`.:..,,,,:::,,,:::::::::::::::;;:;;;;;;''##############
''''''''''''''''''''''''++++++############@@##@@####@@+##@#@@@@@@@@#+'''''''''''''''''''''''''''+++++`  ` ``  ````   ````.```````````````...````````,,,:,,,::,,,:::::::::::::::;;;;;;;';';'#############
''''''''''''''''''''''''+++++++#######+####@######@@@@###@@@@@@@@@@@#''''''''''''''''''''''''''+'+++  `` ``  ` ```   ```.```` ```````` ```..``````````::::,::,,,::::::,,:::::;::;;;;;;;;''''############
'''''''''''''''''''''''''++++++#######++#########@#@@@@##@@@@@@@@@@@#+'''''''';`;'''''''''''''''+''  `` ``   `   ` ``````         ```````` ````````````,;:,,:,,,::::::,,,::::;;;;;;;;;;;'';;'###########
''''''''''''''''''''''''++++++++###++;'+#######@#@#@@@@@+@@@@@@@@@@@@+''''''';``:'''''''''''''''''```` `` ``  `  ` ```            ````````````.``````````;:,,:,,,:::::,,,:::::;;;;;;;;;;;''';'##########
''''''''''''''''''''''''''''+++++###;':'+####@##@@@@@@@@@@@@@@@@@@@@@#''''''':``:;':'''''''''''';````     `   ```                ````````````..,`.`````.``;:,::,,:::::,,,:::::;;;;;;;;;;;;;'';;#########
''''''''''''+:''''''''''''''++++'';,;;','########@@@@@@@@@@@@@@@@@@@@@+''''+',```.,;'''''''''''; `   `   `   ``                `` ````` ````..``,,.`````.``::,:,:::::,,,,:::::;;;;;;;;;;;;;;''''########
'''''''''''+,`;''''''''''''++'++#;',,:;':++#####@@@@@@@@@@@@@@@@@@@@@@#''':,``.`` ,'''''''''''' `` ``   `   `       `` ``        ```````..```,``.``.````````:::::::::,,,,,::::;;;;;;;;;;;;;;'''''#######
''''''''''':`.:';,''''''''''''+++'::,:'':'+######@@@@@@@@@@@@@@@@@@@@@@+''+.```.``+'''''''''''` ` ``   `   `           `      ```` ``````..``..````````````.`.;::::::,,,,,::::;;;;;;;;;;';;;''''';######
''''''''''':...`::''''''''''''''++':,,;'':+######@@@@@@@@@@@@@@@@@@@@@@@'''+`````''''''''''''' ``` `  `               ``         `    ```.,,,`....```````````.,;;:;::,,,:::::::;;;;;;;;;;;;;;''''';#####
''''+'''''``...`,'''''''''''''''+#;,:::;';#######@@@@@@@@@@@@@@@@@@@@@@@#''''` ``''''''''''''`````` ```                          `` `````..`,:,..````````````.`;;;:;,,,,,::::::;;;;;;;;;;;';;;;'';';####
'''''''+,,```..`+''''''''''''''+++'':::;'++#######@@@@@@@@@@@@@@@@@@@@@@@''''```,''''''''''';`````` `` `                            ````````.:;.````````````...`;+:::,,,::::::::;;;;;;;;;;;;;;;;;;'';###
'''++'+''`````.;'''''''''''''''+++'#'+;:'+#####@@#@@@@@@@@@@@@@@@@@@@@@@##''',` :'''''''''''```` ,`` ``` `          `   `     ` `````````````.::,`````````````,.`'++:;:,:::::;::;;;;;;;';';;;;;;;;''';##
+'+''++'+:`````''''''''''''''''++'####+'+#########@@@@@@@@@@@@@@@@@@@@@@@@+'+,:,,:''''''''',```````` `.``          ``    ```  ````..,,...`````....`````````.```.``++#':;;;:;;::,:;;;;;;;;;;;;;;'';'';';#
'++''+'''+ .``:'''''''''''''''''++#####''########@@@@@@@@@@@@@@@@@@@@@@@@@#'',:+.,'''''''''```.`. ``` ``  `          `..  `  ``..:;:.....:......`....```.`````.....++++';;:;;:::;;;;;;;;;;;;;;;;';;';'';
'+''++''''; `:,'''''''''''''''''+#######;##########@@@@@@@@@@@@@@@@@@@@@@@@';,+'+.'''''''':`.`.`. ````.   `        ``.`  `````.::,'':`.`.`,.......`,.`.`...`.`.`..`:++++#';;::;:;;;;;;;;;;;;;;;;;;;;;'''
+''++++'',,,+,:'''''''''''''''''###################@@@@@@@@@@@@@@@@@@@@@@@@#,'''''''''''''`... `.`````.``       ```````   ```,';;:`:+.,,```.,,,,...`......```.....``'++++##;;::;:;;;;;;;;;;;;;;;;'';;'''
++'+++'+',,+'+,''''''''''''''''+####################@@@@@@@@@@@@@@@@@@@@@@@@''''''''''''''`..``````````````     ` `````````:'+++'';:,:`,,.,..,,.,,..`....`..:,..`````++++#+#+;';;;;;;;;;;;;;;;;;;';';;''
''''+++'+,+'''+''''''''''''''''+####################@@@@@@@@@#@@@@@@@@@@@@##+'''''''''''';.,```. ``````` `     `  ```` ``.:###+#++';;;:,..,..,:,,,,.`.,,,....````````.++++++##+;;;;;;;;;;;;'';';;'';;;;'
''+''+++,+'''''''''''''''''''''+####################@@@@@@@@@@+@@#@@@@@@@@@#@'''''''''''',,.``.``````````` `   `````` `.::#++'++++'';;;;;,,,,,::::,....``......````.``'+++++####';;;;;;;;;;''';'';;;;;;;
''''''++++''''''''''''''''''''+++####################@@@@@@@@@#+###@@@@@@@@#@+''''''''''',..`,,.`.`.```````` ````.````,.'#++'+'##++';':;;;;;'';'''::,,...,:,...........+++########;;;;;;;;;';';;';;;;;;;
'''''''++'''''''''''''''''''''++#####################@@@@@@@@@@#++#@@@@@@@@@@@''''''''''':,`.::,..`:```.```````.```...:##++'''''#+''';;:'''':;:;:::::...,.,,,...`.....`'+##########;;;;;;;;';';;';;;;;;;
''''''''++''''''''''''''''''''++#####################@@@@@@#@@@#++#@@@@@@@@@#@+'''''''''',,.,,::,``;`...`````.....`,,;+#+'##+';;##+;;::;;:;:::,,,,,.....,,,,,`..```.``.`+###########+;;;;;;';;';;';;;;;;
'+'''''''''''''''''''''''''''''+######################@@@@@@#@@#++##@@@@@@@@@##'''''''''',::...,:.```.,`````.,,.::;'+##+++#`@';'##+::;'',,:,,.:,,,,::,,,,..,.....```.``.:#############'';;;;;'''';'';';;
+''''''''''''''''''''''''''''''+######################@@@@@@#########@@@@@@@@#@''''''''''::`````,,` ``....`.::;;;;'++##+++###+';+#+:::;;.:.::,,`,.;:.,..`.:,,....``.`.```+###########+#+';;';''''';';;';
++''''''''''''''''''''';''''''++#######################@@@@@@##@@####@@@@@@@@@@'''''''';;` `````.``` `.:`..:;''';'''###''+###;;;+#':,,'.,.,,,,.,.,,`:.,,,,.....,.```...`.;###############';;;'';'''';;;'
##+''''''''''''''''''''`,'''+'++#######################@@@@@@@#@@@+###@@@@@@@@@+''''''''.`` `````.```..:,:,;''';;'''###'''###;;;##::,::`,,:.,,,.,..,,,,...`.`.....````````################+;;''';'''';;;
#++'''''''''''''''''';`..;;,,'++##################+'####@@@@@@@@@@@+#@#@@@@@@@@#'''''''..````` ```.`...:;,;;';';;;;;+##+'';';;;;#+:::'.`,::,:...,.:.,...........```````.``:################+;''''''';';;
++++''''''''''''''''+,`.` :,+''++###########+####+:;:####@@@@@@@@@@##@@@@@@@@@@@+''';'.`````     ````.,::;;;;;';;;;;;+##''''';;+#;:,:;`.,.::,,,.,,..,..,........````.``..``#################+;''''''';;'
'++++'''''''''''''';.``..`.''''++############;;'+'.;;'####@@@@@@@@@@#@@@@@@@@@@@@''''.````      ``.`..,:::::;;';;:;;;;'##::;;:'#;:,,'.::.,..,,,,,.,,.,,,......`,,..`````,``'#################+;'''''';''
+''+++'''''''''''',:`````.:''''++###########++;;;:,,;++###@@@@@@@@@@@@@@@@@@@@@@@+'';`````    ````,.`..;::,:;;;;;::::'';##',;++;:,:;:,...,,,,,,,.,,,,,,,,,,...,,.`.```.`,.`,##################';'''''';'
++''+++'''''''''''',`````,+''''+++##########++';,:::,''+###@@@@@@@@@@@@@@@@@@@@@@#''.``````````.......`,:,,,::::::,::;';;'###+;;:,,:.`..,.,,.,..,:,...,......`.,,..````....`###################';'''''';
''++'+++''''''''''';.``` +''''''++##########+++;,,:,.''+#@#@@@@@@@@@@@@@@@@@@@@@@@';.`````````..```,...,:,,,:::,,::,,,''';;;;;;:;;:``..,,,,,.,,:,.,...,,,.....,.`..``````.:.'###################';''';''
++''''+++''''''''''+````,'''''''+++#########++++,.,::;''##@@@@@@@@@@@@@@@@@@@@@@@@+...````` `,,..``.....,,,,::,,::,:,.,;''';:,,;;:..:,:,..,,,,,..:,,,,,,.......,....``````:.:####################'''''''
++'++''+++'''''''''+:` :,'''''''+++#########+++;;.::;:+###@@@@@@@@@@@@@@@@@@@@@@@#:```` ````,..,..```,...,.,.,.,,,,....:;;;'+'';,....`,.:,.,,,..:,,.,,:,....`..,,....``````,.#####################''';;'
''''''''++''''''''';::#,:;'''''''+++#######++++;'+.;;:+###@@@@@@@@@@@@@@@@@@@@@@#;``  `````.``.:,,...,`.,.....`..,..`.,,,:,.``..`.,,,,,,,,,:..,;,,.,::,,......,,,:,.`.`````..+#####################'''''
'+'+'''''++'''''''+,,+'+,,'''''''+++#######+#+';'+#:;;####@@@@@@@@@@@@@@@@@@@@@@+```` ``..,;:,.,:,,....`......````.`.``.,..``...,,,,,,,,,,,,.,:,..,::::,....,....`...```````.'#####################+;';'
''''''''''++'''''';:+'''+:'''''''+++#########+''+#+#:'###@@@@@@@@@@@@@@@@@@@@@@#..``   ...###';,.,,...,.....``.`````.`....,.,..,..,...,,,:,,,,,..,,:::,,,,:,......,,`````````:######################+;''
+'''''''''''+'''''''+'''''''''''+++++########+++++#+#;###@@@@@@@@@@@@@@@@@@@@@@;.``    `..++##+;..,...`,.......`````.`.`..,.,.,..:,.,:,,:,,:,,,,,::::;:,,,,:,,,,,.`..````````.#######################+;'
'+'+''''''''''''''''''''''''''''+++++#######+++#++++##+##@@@@@@@@@@@@@@@@@@@@@#.```     ...,'+++.......`,...`.````.`...,,..,,,.,,..,..,,:,,,,,,.,,:;;;:,,........,,...```` ``.########################+;
'''''''''''''''''''''''''''''''''+'++##+##+++++#+++#+####@@@@@@@@@@@@@@@@@@@@@;.``````````   ..;:...```.````.`.`````.,..,.,,,,::,,,.,,,..,.,.,,,::::::,,,..,,,........````````;########################+
'+''''''''''''''''''''''''''''''''''+++++#++++'+#+++#####@@@@@@@@@@@@@@@@@@@@#.````````` ``  `  ,.....`.....``````...`...,:,:,,.,.,,,..,,,:,,:,,,,,::,,,..,,,,,..,...``` `````.#########################
''+''''''''''''''''''''''''+++++++++#++#++++++'+++'+######@@@@@@@@@@@@@@@@@@@;.`````````` `` `  `.,.``.``..`````.`...,.,.,::;;'';:::,:::::,,...,,,,,,..,,,,,,......,...````````#########################
+''''''''''''''''''+++';:.```.````..,,.:::;'+++++#++++#####@@@@@@@@@@@@@@@@@@.``````````  `.`   ``.`.``..``....,,,,.,::,,;;;;:,`.;:,,,:,,.,,,,,,,,,,,,,,,:,,,..........``````.`;########################
'''''''''''++++';:,.````.....``......,,,,.:,.,,...,,;'+#####@@@@@@@@@@@@@@@@+.`````````````.,,.`...``.``.....`,.,:,;,:,::,,.`````.,:,,...,,,:,,,,,,::,,,:,,,,,,,....```````````.########################
'''''''+++;.```.......```..`````.`...,,,..,,,,,..,.,,,,:;##@@@@@@@@@@@@@@+@@:.```````````````.,..,......,:::,::,,,,.` . `.`....,,``..`..,`,.,,,,,,,,,,,,,,:,,,,,.....``.`.````.`########################
'+#+;;,......`.``..```````.``````....,,.,..:,..,.,...,,,,:;#@@#@@@@@@@@@@###..``````````..``  `.`...`.,:,,...```````.,.,..,,:;;:``.``````..,.,:,,,,,,,,....,,,,,,,....````. ````'#######################
,.``,...,``..````..```````.```````....,,,...,........,,.,::;;@@@@@@@@@@@@@#'.``````````....`````....`....`` `..``.:'+++++++##;.``,.````...,,,,,,,,,,,,.,,,,,,,,,..,,.````.`,```.,#######################
.....,...`````````````````...````.....,,,...,,...`...,,,,:;:;:#@#@@@@@@@@@#:..````````....``````....``````..```,;+''''++++'.  ``,,```....,,...,,,,,,,,,,,,,.,,,,..``..`,.``.`````#######################
...`.....`````````.````````..`````...,,,,,..,,........,,,,:;;;;#@@@@@@@@@@@,.`````.``....``..`..`.`.```.````.:'++++++++':` ````,,``````,,,..,,,,,,,..,,:,:::,,,..,.``....````````#######################
``........`````````````````..`````...,,,,,,..,,........,,,:::;;;@@@@@@@@@@#,.`````.`.........````` ```````,;+++++###;. ``` ``.::``....,........,,,,,,,,,:,:::,,....,.`..`.`...```'######################
``......`.`````````````````...`````..,,,,,,..,,........,,,,:;;;;;@@@@@@@@@'..````............`.`````..``,'++#++++#'```   ``.,;:..`....`...,,,,,,,,,,,,,,,,,,....`...`.`.`..` ````:######################
```.......``````````````````..`````..,,,,,,,..,,.......,,,,::;;;;;@@@@@@@@:..```............`..```.,.`.;+####++#+.  ` ``..,;;:..`.......,.,,,,,,,,,,,:,,,,,,..,......``.`` ```````######################
```......```````````````````..`````...,,,,,,..,,........,,,::;;;;;'@@@@@@@:...````.`..........``.....,+#######+:   ```.,:;;:,...,.....,,,,:,,,,,..,,,,,:,,,,:,`.......````.`````` ;#####################
````......```````````````````..`````..,,,,,,...,........,,,::;;;;;;;#@@@@#,...``......,...`..``.,,.:'########:` ```,:::;;,,,..,,,,,,,,,,,:,::,,:::,:,,,,,:,,,::.....`````````.`````#####################
````......```````````````````..`````..,,,,,,,..,.........,,:::;;;;;;;+@@##,...```...,........`,:,:;+##++''+;`  ``.:;;::,,,,..`....,,.,::,::::::::::,,:,,:::,,,.,,.,..````` ` ` ```,+####################
``````.....```````````.``````...````..,,,,,,,............,,,::;;;;;;;;;@@+,........,,,,,,....,::;+#++'''':`  ```,';::,,:,,..,,,,,::,,:,,,::,:::;:::::,:::::::,,.,,,.,``````  ` ````.+###################
`````......```````````````````..`````..,,,,,,..`..`......,,,::;;;;;;;;;;+#,.......,,,,.,,,..,::+#+++''';`  ```.;:::,..,.....,,,,,,,::,,,:,,:::::::::,:,:::::::...,.,....`````     ``:###################
``````.....````````````.``````..`````..,,,,,,,............,,::;;;;;;;;;;;;,.......,,,,,,,,.,,:+++'''';.   ` `:;;::,,,.`...,..,,,,,,,,.,,,,,,,,,::::,:,,:,::,,::,....,,.``````.    ``.###################
``````......``````````.```````...`````..,,,,,,............,,:::;;;;;;;;;;;,,......,,,,,,,,,,,++'';;;,`````.,;;;:,:,,,......,,,,,,,,,.,,,,,,,,,,,,:::,,,,,.,,...,:,.```...`````.`   ``;##################
```````.....``````````.```````...`````..,,,,,,..`..........,:::;;;;;;;;;;;,,......,,.,.,,,,,'+';;;:`  ```,;;:;:;,,,,,...,.,,,,,,,,,.,,,,::,:,:,.,,,:::.,,...,...,,..` ```.,``` `. `` `+#################
`.`````......``````````.```````...`````..,,..,,............,:::;;;;;;;;;;;:,,,...,,,.,,,:::;+';;;.` ``..,;;::,:,:,::::,:,,,.,..,....,,,,,,:,,:,..:,,:,:......,,...,`..  ``.,```` ` ```:#################
...`````.....``````````..``````...`````..,,,.,,..`.........,,::;;;;;;;;;;;,,,,...,,:,,,:::;++'':````...:;::::,,:::;;;;::,..........,.,,,,,..,,,,,,,,,,.,......,,,....`..`````` `  ```` +################
..```````.....``````````..`````...`````...,,..,,...........,,::;;;;;;;;;;;:,,..,,,,::,,::;++'+:`.....,:;:,:;:,;:;;;;:...,..........,,,,,.....,,,,.:..,,.....,..,,.,.```..````       ```,+@##############
...``````.....``````.```..```.`...``````..,,..,,.``.........,:::;;;;;;;;;;:,,,..,,:,,,,:+++:,,`..,..,:;::::::::;;,...`.........`.,..,,...`.`..,,,..,..:.........,.,,..``.,` ``   `  ``` ++@#############
,,.``````.....```````.``...`````...`````...,,.,..``.........,,:;;;;;;;;;;;:,,,,,,,::,:'#++:,.`......,:::;::;:;:,..`...............,,,.....````...,,,,.,:,....,,....,,,,.`...`        `` ;+##############
,,..``````.....``````..```..````...`````...,,.....`.........,,;;;;;;;;;;;;:,,,,,,:;:;+##+;,......,,,:;;;;:;;,,...```.````````...,,,....`````````....,,,,:.....,`.....,,,,.`...        .`.''#############
,...``````.....``````..```..`````..`````....,,.,,.``........,,:;;;;;;;;;;;::,,,,,;::+#++',.....,,,,:;;;;;::::,:....`````````........`.`.`.`.........,,,.,,.`...,...``....,.`.``       `:`'++@###########
....```````....```````..`...`````...````....,,..,..`.........::;;;;;;;;;;;::,,,,,;:'#++',`....,,,:::;';;:,.`.,,:,,.````````..`.......`..`..```..``...,.,.,.`..,...`.``..``..``         ;;''++###########
....``````.....```````.....````......```...,.,...,..`.......,:::;;;;;;;;;;::,,,,::;+#++;`....,:,::;;;;:::,`.```.``.........`.......`..............``...,...`.``....```````````  `      `+'''+###########
.....``````.....```````...``.........``....,,,:,..,.......,.,,;:;;;;;;;;;;::,,,,::++'++,....,,,:;';;:;:::.,.````````........```.....`..........`..``.`.....`...`,..,..` ` `````       ` '++''+#@########
.....`````......``````....`..........,..`........,,,..,.,,,,,,:;:;;;;;;;;;;:,,,,::#+'+;.,..,,,:;;;;;:::,,..,,...```` `......`...```................``````````...`,...,``` ```           .'++''++@#######
..``..````.......`.``....,.``....,:;;;'''+'+'+++++'';::,,.,,,:;;;;;;;;;;;;;:,,,,:;#'+#:,..,,,:;;,`:::,::,:,,,,,....```````.......````...........`...```````````.......:..`   `         ` ''++''+'#######
......````.,..,..`````.,:;'+++'''''''''''''''+++++++'+'+'+++++';;:;;;;;;;;;:,,,,:'+'+'.,.,,,;;;.``.::::::,,,,,,,,.........`..`..`.``````.........`...``````.```..,.,`.,:.``              `''++''+'+@####
........`...``.,:;'+++'''''''''''''+''''''''''++++++++'++'+'+++##+';;;;;;;;::,,,:'++#;,,,,:;;;.````.::,,:,:,,,,,,,.......,..........````.......`.`..`..`````````..,,....,,`.`         ``  ;'+++'++''####
....,.`.,'+++++'''''''''''''''''+''+'''''''''++++++++++++++++++++##+;;;;;;;;:,,,:''++.,,,::;;:,`````,,,,::,:,,,,,,,,..,.,...`.....`````...`.......``.``.````..`...`,.``...,```     `    ` `''++''+'''#@#
...,;'++'''''''''''''''''''''''''+''+'''''''''++++++++++++++++++#####';;;;;;:,,,:''+,,:,::;;;:,`````:.,:,::::,:,,,,,.,,,``..........``````.``.......```..`..,.```..`..```````           `` ;''+++'+''''#
++++''''''''''+''''''''''''''''''+''++'''''''''++++++++++++++++++######;;;;;::,,:'+'::,::;;;;;..````,,`,::,::::,:,,,,.,,,.............```..`...`...``.``..``,..```.``.`.``               ` `''+++'''''''
+++'''''+'''''++''''''''''''''''''''++'''''''''++++++++++++++++++#######';;;::,,;'+:::::;;;;;;`.````.:``.:::,::::,,,,,.,,,,..............```..`...`...```,..`,,,```..`` .`               ```'''+++''''''
++''''''''''++'++''''''''''''''''''''+'''''''''++++++++++++++++++########+;;;,,,:;;;:::;;;;;;;,..````.```,,::,,::::.:,,,,,,,......,.......``.....,.``..``.,.`.,,,..`..    ``             `` ,'''+++'''''
++++'''''+''+++'+'''''''''''''''''+++''''''''''++++++++++++++++++##########;;:,,;;;:;;;;;;;;;;;`..````````,,,,,,,::,.,,,,,,,............,.`.``....,.`..```,:`...,.,``.`    ```           ````''''+++''''
++++''''''+'+++''+''''''''''''''''++'+''''''''''+++++++++++++++++##########+;;,,;;::;;;;;;;;;;;```````````...,,,,,::,,,,,:,,,..............`...,...,``..``.,,..,,,,.```.     ``` `     ``````''''+++''''
++++''''''++'+++'++'''''''''''''''+++'+'''''''''++++++++++++++++++##########+;:::;;;;;;;;;;;;;;`````..```````,,,,,,,,,,,..,,.,,....,...........,,...,``..``.,...,:,,.` `.       ``      ``````''''+++'''
++++''''''+++++++'+'++'+'''''''''''+++''''''''''++++++++++++++++++###########';,;;;;;;;;;;;;;;;:````...```.``..,.,,,,,,,...,,,,.....,,..........,,,.,.`...```:,..,,,,,`  `        ``    `````.;'''++++''
+++++'''''''++++++'+'++++''''''''''+++'''''''''''+++++++++++++++++############;;;;;;;;;;;;;;;;;;````....``..``.,.,,,,,,,....,,,,.,.,.,.......`..,,,,.,.```.`,.,...,,,,..                 `` `.,''''+++''
+#++++'''''''+++++'+++++++''''''''''+++'''''''''+++++++++++++++++++###########;;';;;;;;;;;;;;;;;````.....``.``,,..,,,,,,,,..,,,,,.,........`.``.,,,,....``..`.,,.,....,..                `` ``,''''+++''
##++++'''''''++++++++++++++'''''''''++++''''''''++++++++++++++++++++##########+;;';;;;;;;;;;;;;;```.......````,,..,,,,,,,,,..,:,,,.,..........`..,,,.,,...`...,.,`.,...``.                ```..,'+''+++'
##++++''''''+'++++++++'''++''''''''''+++''''''''+++++++++++++++++#++###########;;;;;;;;;;;;;;;;;:```.......``.,,..,,,,.,,,,,.,::,,,,.....,....``..,..,.,..`..`.,..```..`` ``              ```.,`:;'++++'
###+++''''''+++++++++++''++'''''''''''++''''''''++++++++++++++++++#+###########';;;;;;;;;;;;;;;;; ```.......`:,,,.,.,,,,,,,.:,,,:,....,........`..,.,,.,.,``....,`..``.``` ` `            ` .`:;,:;''++'
###+++'''''''+'++++++++'+'+'''''''''''+++'''''''++++++++++++++++++##############;;;;;;;;;;;;;;;;;.``.........;:,,,,,,,,.,,,:,,,,,,,....,.....`..`.,..,,,,,,.`.``.`  .``,`````              ` ..:;;:;++++
####+++'''''''+++++++++++''''''''''''''+++''''''++++++++++++++++++##############+;;;;;;;;;;;;;;;;,``.`......,;:,,,,,,,..,,,,,,,.,,,...,.......`.`.,`,,,,,...`..``.  `.  .``````            ```.:::::;'++
####+++'''''''+++++++++++'''''''''''''''++''''''++++++++++++++++++###############;;;;;;;;;;;;;;;;.```.`..,..,::,,,,,.,,,,,,,,,,,...,.................,:,.,.,......`   `  `` ````           ```..';::;;'+
####+++''''''+'+++++++''+'''''''''''''''+++'''''++++++++++++++++++################';;;;;;;;;;;;;;`.```.......::,.,,,,,,,.,,,,,,,,.,,,`.,.,.........,.,,:,,,.,.`..`,`         `` `    `       ``.`,:::;;'
#####++''''''+'+++++++'''''''''''''''''''++''''''+++++++++++++++++##################+';;;;;;;;;;;,..``....,..:,,.,,,,.,,,,:,,,,,.,..,.`.,,.,........,.,:::.,,.``....`        `````           ```.,:::;;;
#####+++''''''+'++++++'+''''''''''''''''''++'''''++++++++++++++++++##################@+;;;;;;;;;;;....`,...,.:.,,,,,,.,.,,,,,.,,.......`......,....,.,,,,:,,,..``..,.          ```            ``,,,:::;;
#####++''''''''+++++++++''''''''''''''''''++'''''++++++++++++++++++#####################+;;;;;;;;;,`...,,.,,,;`,,,,,..,.,,,,,,,.........`........,.,...:,,:,,,..`.,..``        ````           ```:,:::;;
######++'''''''++++'++++'''''''''''''''''''++'''++++++++++++++++++++############################+#'''';,,.,.::.,,,.,,.,,.,,,,,.,.......`........,..,,..,,,:,.,...`,:..``           ``        ```..:::::;
######++''''''''++'+++++'''''''''''''''''''++'''++++++++++++++++++++############################+#''+''++++';,`.,,,,,.,.,.,,,,,.,......``....,``....,`.,,,.,,.:,.``.:..`.`                     .`,,:::::
######++''''''''''++++++''''''''''''''''''''+''+++++++++++++++++++++#############################+''+'+++++''++:,,,,,.,,..,:..,.:.......``....`.....,.`..,,.,..,.``..,``.``             ` `  ``.,::,::::
#######+'''''+''''++++++'''''''''''''''''+'''++++++++++++++++++++++++#############################'''++++++''++:,,,,,.,...,:..:,.,..``..``...`.....,.,`.`....,..,` ...,..```              ``````,,::::::
#######+'+''+''''+++++++''''''''''''''''''+''++++++++++++++++++++++++#############################++'++++++'+++:,,,,,.....,:.....,..``..``.....`...,.,..`.....,,`. `.`.,.``              `    ``..:,::::
#######+++''+'''++++++++'''''''''''''''''+'''++'++++++++++++++++++++++############################+++'+++++++++:,,.,,,,....,,..,....``.`.`...........,.````....,.`` ``..,.```                  `:;:,,,::
########+++++++'+++++++++''''''''''''''''''''''+'+++++++++++++++++++#+#############################++++++++++++:,:.,,.,...,.:.,,....`....`....`.......,`````....,`.``` .`.` ``               ````;':,,::
#######+++++++++++++++++'''''''''''''''''''++++++++++++++++++++++++++++############################++++++++++++:,:.,,,,...,`:.......``...``.....`.....,``````````. ` `` ` ````            `   ````;,,,,:
;'#####++++++'++++++++++'''''''''''''''+++''::,,..,,,,::;''++++++++++#+############################+'++++++++++:,:.,,.,...,.,.......``..```..`.,...`..,.````````. ``   `   ````      `         ```:,,,,:
;';'+##++++'''++++++++'+'''''''''''+';:.````.`..,,..,.,,,,,,.,::;;++++++###########################++++++++++++:,:,,,.,,....,.......```...`..``,.`......`` ` ```````         ``   `` `        `````:,,,,
;;;'';;;'++++++++++++++++''';;:::,``````````````.......,.,,.,.,,,,,:,;'+###########################++++++++++++,,:.,,.......,.`.....```..``..`..`.....,., ``` `..```           `   `   `   ``  ````,,,,,
;;;;'';;;;:::;:,::,:,:,.,...,..`.```````````````........,...,,.:,,.,:,:,:;+#########################+++++++++++,,:,,,..,............````.```.`,......`.,``` `````````               `   `  ``` ```..,,,,
;;;;;'';;;::;:,:;,,:,,.........`````````````````....`...,,,..,..:,..,::::::,'#######################+++++++++++,,:.,,,.....`,.`......```````..,........,`, ```````````              ```  `````````..,,,,
;;;;;;;';::::,,:.,,,..........``````````````````.........,,...,.,,,.,:,:,:,::::######################++++++++++,,::,........,.`......````.`..........``.,. ````````````            `` ```````..```.:.,;;
;;;;;;';':;:,,,.,,,....```...``````````````````..........,,,..,,.,,..,,:,,:,:;:;'####################++++++++++,,:,,...,........,`..`````.``.........``..`.` ```````````            ``````````.``..;''''
;;;;;;;;;;:,,,,..,.....``....``````````````````......`...,,,...,.,,,.,::,,::::;:;;+##################++++++++++,,:,,...,....,.......``.``.`...`.`.......... ` `````````````         `` ```.````.`:':'+''
:;;;;;;;;;;,::.,......```...```````````````````.....,.....,,,..,.,,,..,:::,::::;:;:'##################+++++++++,,:,,,.,,.,..,...,`..````..```..,..,......`..``````````````````   `  ```````.`.`.,.'+++'+
::;;;;;;;;;::,,,,.....```....``````````````````......,..,.,,,..,,.,,,.,:::,,:::;;;;;'#################+++++++++,,,,,,.,,...,,.,.,....`.``.`.....``.....`.`.:```````````````````````````.`....`.,:.,+++'+
::;;;;;;;;;;,,,,...,..`.`.`..````````.`````````......,...,.,,,..,,,,,..,:::,::::;:;;;'#################++++++++,,,,,,..,..,,,......,...`..`.....```......``:`````````````````````````..,.......,'::'+'#+
:::;;;;;;;;;:,,,.......`.``..````````.`````````.......,....,,,..,,,,:,,,:::,,::::;;;;;'################+++++++':,,,,,..,..,,,...,..,.`.`..``....`........`.`;``.```````.`````````````..,,....,,,:+''+#++
:::;;;;;;;;;;,,,,.......``.```````.````````````......,,.....,,,..,,,,:..::::,::::;;;;;;+################++++++;:,,,,,..,..,,,,,...,,....`.``......`.....,..`;```,``````..```````````...,:....,:;'+++##++
::::;;;;;;;;;,,,,.......```````` ``````````````.......,....,,,,..,,,,:,,,::::::::;;;;;;;################++++++;;,,,,,.,,,,,,,,,...,,,..`..,`.....``.....,,.`.:```,``````...`````````...;::,,,.,++##+#+++
::::;;;;;;;;;:,,,......``````.`````````````````......,.,....,,,..,,,,::.,::::::::;;;;;;;;################+++++;;,,,,,.,,,.::,,,,..,,.,.``.,......`.`....,,..`,.`.`.`.````..`````````....;':,,,.,++++++++
:::::;;;;;;;;:,,,,.....`````````````.`````````.......,.,......,,..,,,,:,,,:::::::;;;;;;;;'################+++':;,,:::,,,,,:,,,,,,.,:,,..`.,..,....,`....,,:,..,``.,``..````````.````...,:';;:,:::;++++++
:::::;;;;;;;;::,,,.....`````.```````.``````.`........,,.......,,..,,,,::.,:::::::;;;;;;;;;+###############+++':;:::,:,,,,,:,:,,,,.,;:,,.`.,.,.....`.....,,:,..,`..`:.....`...``........,:+#';:;;;;++++++
::::::;;;;;;:;::,,.....```````````````````............,.,.....,,,..,,,,:,,,::::::;;;;;;;;;;################+'':;:::::,:,,,:::::,,,:::,,.`,,.,..,...:`...,,,:,...`...,.,................,;'+;:::;;'++++++
::::::;;;;;;;;:::,......```.````````..```.............,,......,,,..,,,,::,,:::::::;;;;;;;;;'###############+'':;::;::,:,:,::;::,,,::;,,..,,,,,.,,...``.,,,,,,.``.,.,.,,,,.`............,,;:;::;;''++++++
::::::::;;;;;;:::,,......`.`````````...``.............,,.......,,,..,,,,:,,:::::::;;;;;;;;;;###############+'':;::;;:::,:,,;'::,,::::,,,.:..,,,.,,..:`..,,,,,..`.,.;....,..............,:,:;;'''''++++++
:::::::;;;;;;;::::........``````````...`...........,...,,......,,,,.,:,.::,,:::::::;;;;;;;;;'###############'':;::;;::::,,::+;;,,;:;:,,,,:.,,,,.,,,.:`..,,,,,,...,.:.....,,.........,.,,,:;;;''''''+++++
::::::::;;;;;;;;::,......```````````....`..............,,,...,..,,,..:,,::,,:::::::;;;;;;;;;;+##############+':;::;::::,,::;#;;::':;:,,,,:.,:,,.::,.,.....,,,,,..,.:,,.....,......,.,,,:::;;;'''''++++++
:::::::;';;;;;;;;:,.......`````````.........,,......,,.,,,,...,.,,,,.,:,,::,,:::::;;;;;;;;;;;;##############+':;::;:;,;::;;+#;::;;:;;,,,,,,,,,,,,,,,.:`.,..,,,,.,,.::,,.....,,.,,,,,,,;;:;;';'''''++++++
:::::,:;;;;;;;;;;::,......`````````...,.....,.......,,..,,,,..,,,,:,..:,,::,,:::::;;;;;;;;;;;;'#############+;:;::;:;,;;;;:##;:,';;';,,:::,,,,,,.,:,.:..,..,..,,,,,.;,,,...,.,,.,,,,::;;;;;;;;;;''++++++
:::::,:';;;';;;;;;:,.....``````````.`,,,....,,.......,,.,,,,..,,,,,:,.,:,,::,:::;;;;;;;;;;;;;;;+############+;:;;,;:;,;;:':##';:;;';;,,:;,,:,,,,,:,,,.;..,.,.....:,.,:,,,.,,,,,:,,,,:;''':;;;'''''++++++
::::::;;;;;;;;;;;;,,.....```````````..,,.....,.....,.,,.,,,,,..,,,,::..::,::,,::;;;;;;;;;;;;;;;;''+########++;:::;;::,'':':##+;:''+';:::;,,,:,.,,,,::.:,,,,,...,,.,,,::,,.,,,,,,:,,,:;'';'''''''''++++++
:::::;;;;;;;;;;;;:,.,....``````````...,,......,..,..,.,,.,,,,,.,,,,,:,.,:,,:,,:::;;;;;;;;;;;;;;;';;;;;''+##++;::::;::,;+'+:###;;'++';:;:;:,::,,,,,:,:,,::,,...,,....,,;:,,,,,,,,:,,;;;';;'''''''''++++++
;;;:;;;;;';;;;;;;,,,,,...``````````...,,,..,..,..,,,,,,,.,,,,,.,,,,,::,.::,::,:::;;;;;;;;;;;;;;;;;;';'''';;;;:::::::,:;###;###':'#+';:;:;:::,,,,,,;:::,:;,.,,..,.,,,,,:;::,,:,:,,::';;;;;';;'''''+++++++
++'';;;;;;;;;;;;,,,,,....`````````.....,,......,..,,.,,,,.,,.,..,,,,,:,.::,,:,,::;;;;;;;;;;;;;;;;;;;';'''';;;:::,:,::;'###'####:+##'':',:::::,,:,,:,::,:;:,,,,,,,,,:,,,;::,,,:::::;'';'''';''''+'+++++++
#+####++';;;;;;:,,::,.,.`.``````..``.,,,,,..,,.,,,,,,,,,,,,,,,,.,,,,,,:,,::,::,:::;;;;;;;;;;;;;;;;;;;;''''';;::,,:,::;:+++'####'###+':',:;:::,,:,,:;::,,;;,,,,,,,,,,:,,:;::,::::::''+++''''+++'+++++++++
#######++#++'';:::::,,,....``.`......,,:,,,.,,,,,,,,,,,,,,,,,,,,,,,,,::,,::,,::::;;;;;;;;;;;;;;;;;;;;;''''';;::,:,,;:;:+++;++++++###';;,:::;:,,:,,:;,;,,;;:,,,,,,,,,,,,::;::::';:''++++'++++++'+++++++++
########+#++++++#+';;:,.,.......,..,.,:,,,:,,,,,,,,,,,,,,,,,,,,,,,:,::::,,::,::::;;;;;;;;;;;;;;;;;;;;;''''';;,:,::,:;::++''++++++++';:;::::;:,,:,::,;:::;;;,,,,,,,,,:,::::::::'';''+++''++++++'++##+++++
###########++++#+++++++''';;;'''''''+'+++++++++++++++++++++'+++'''';;:::::;;::;;:;;;;;;;;;;;;;;;;;;;;;''''''',;:,:::'::+'+;+++++++''':,.,,::,..,,:::',:::;;:,:,,,,,,:,::::;:::'+''+++'+''++++++++###++++
############+####+++++++'+'''++++++++++++++++++++++'++''+''''''''+++'+++++;;:::;:;;;;;;;;;;;;;;;;;;;;;;'''''':;:::':+;;+'';+'+'''+'+':.:,,::,..``.....,..,,:.``...,,:::::::;::'+''++'++++++++++++####+++
++################+++++++++'++++++++++##++#++++++++++''+'''++''+''++'++++++++++';;;;;;;;;;;;;;;;;;;;;;;'''''':;:::+:+':'++''''''''''+:,:.,,:,,.``.,..:.,..,:,````......,,::;;;;'+'+'+'+++++++++++#####++
#+##################+##++++++###++#++##+++#++++++++++'++'''++'+++++++++++++++++++#+';;;;;;;;;;;;;;;;;;'''''''::,,:+:++,''+''''''''''';,::,,;,,.`..,.,:,.,,.,:.`.......,,,,,,,,:'++'+++++'++++++++#####++
++######################++++############+###++++++++++++''++'++++++++++++++++++++#++##+';;;;;;;;;;;;;;;''''';:::::+;++,'''''''''''''':,::,:;:,.`.....:,`.,,,,,`......,.,,,,,,;:';;;''++++++++++++#####++
++########################################++++++++++++++''++''++++++++++++++++++++#+#####+;;;;;;;;;;;;;;'''';:::::+''+;''''''''''''''::,,:;;:,.`.......`..,...`....,.,.,,,,,,;;';;;;;;;'++++++++++#####+
+#+#################################+#++#++++++++++++++''+++''''++++++++++++++++++#########+;;;;;;;;;;;''''';:::::++'++;''''''''''''':,:,:;'::......`..`..,........:.,,.,,,,:''';:;;;';''';'++++++++####
##+##################################++++++++++++++++++'+++'''''+++++++++++++++++##+#########+;;;;;;;;;''''':::::;++;++:''''''''''''':,::;';::,..,..`...`.,........:.,,,,,,,,';;;;;;;;;''''';;'+++++++#+
##++################################+++++++++++++++++++++++'''''+++++++++++++++##+#############';;;;;;'''''';::;:;++'++''''''''''''''::,:'';;:,.,,.,...`.,,,.......:.,,,,,,,,';;;;:;;;;'''';;;;;;'+++++'
+##+##############+++##############++++++++++++++++++++++++''''''+++++++++++++##+################';;;;'''''';::;:;++'+++'+'''''''''''::,;'';;;:.:,,....`..,,,......;,,,,,,,:,;;;;;:;;;;'''';;;;;;;;;'+'+
#################+#+++++++++######++++++++++++++++++++++++''''''+++++++++++++++#+#################+;;''''''';::;:;+++++++++';'''';''':::;;'';;:,:,,.,.....,:,.,..,,,:,,,,,,:,';;;;;;;;;;;;;;;;;::;;;;:::
#################+#+++++++++++++++++++++++++++++++++++++++''''''+++++++++++++++#+###################'''''''';::;:'++++++++++''''';;;;:::;;;';;;,:,,,,..,..,:,,,..,,,;,,,,,:::;;;';:;;;;;;;;;;;;::,::;;;;
###################++++++++++++++++++++++++++++++++++++++'''''''+++++++++++++++######################+'''''';:;;:'+++++++++++'''';;;;:,;;;;';;::::,,,..,..,;::,,.,,,;,,,,,::;;;'';;;;;;;;;;;;;;:::,,,:;:
##################+++++++++++++++++++++++++++++++++++++++'''''''''+++++++++++++########################+''''':;;;'++++++++++++;;;'';;::;;;;;;;;::;:,,,.,..,:;:,,,,,;::,,,,:;;;;'';;;;;;;;;;;;;;::,,,,,::
##################++++++++++++++++++++++++++++++++++++++''''''''+++++++++++++++#########################+''';:;'';'++'+++++++++;;';;;::;;;;;;;;::;:,:,,,,,:;;;,,,,,;;:::,::;;;;';;;;;;;;;;;;;;;:::,,,,::
###################+++++++++++++++++++++++++++++++++++++''''''''+++++++++++++++#####################+#####'';;;'';+++''+++++++++;;;;;;:;;;;;;;;::;;,,:,,,,,;;;:,,,,;;;::::;;;;;';;;;;;;;';;;;;;:::,,,.,;
####################+++++++++++++++++++++++++++++++++++++''+'''''+'+++++++++++++###########################+;;;'';''''''+++'+'+++;;;;;;;;;;;;;;::;;:,,,,,,:;;;;,,,,;;;::::;;;;;'';;;;;;;;';;;;;::,,,,,,.
#####################++++++++++++++++++++++++++++++++++++'++'''''+++++++++++++++############################+;'''''''''''++'+''+++;;;;;;;;;;;;;:::;;:,,::,;;;;;::,:;:;::::;;;;;'';;;;;;;;;';;';;;,,,,,,.
######################++++++++++++++++++++++++++++++++++'+''''''''+++++++++++++##############################+'''''''''''++'+'+++++;;;';';;;;;;::;;;;:,,,,;;;;;;,,:;:::::::;;;;';;';;;';;;;;;;;;;,,,,,,.
######################++++++++++++++++++++++++++++++++++++''''''''++++++++++++++#############################+#+''''''''''''''''++'+;;'''';;;;;;;;;;;::,:,::;;;;,::;::::;;:;;;;;;;';;;';;;;;;;;;;:,,,,,.
+######################+++++++++++++++++++++++++++++++++++'''''''+++++++++++++++#################################'''''''''''''+''+''';;'''';;;;;;;''';:,:,::::;;::::::::;;;:;;;:;;';;;;;;;;;;;;;;:,,,,,.
+#######################++++++++++++++++++++++++++++++++++''''+''++++++++++++++++#################################+''''''''''''''''''';;''';;;;;;;'''';:::::::;;:,::::::;;;;;;;;;;;;;;;;;;;;;;;;;:,,,,,,
+########################+++++++++++++++++++++++++++++++''''''++++''++++++++++++####################################+'''''''''''''''''';'''';;;;;'''''';::::::::;::::::::;;;;;:;;;;;;;;;;;;;;;;;;::,,,,,
++########################+++++++++++++++++++++++++++++++''''+'+++''+++++++++++++#####################################'''''''''''''''''';''';;;;;;'''''';;;;;::::::::::::::::;;;;;;;;;;;;;;;;;;;;::,,,,,
+++######################+#++++++++++++++++++++++++++++''''''++++++'++++++++++++++######################################+''''''''''''''+;;'''';;'''''''''';'';;::::::::::::::;;;;;;;;;;;;;;;;;;;:::,,,,,
##########################+++++++++++++++++++++++++++++''''''++++++'+++++++++++++#########################################+'''''''''''''';''''+;;''''''''''';'';::::::::::::::;;;;;;;';;;;;;;;;;:::,,,,,
##########################+++++++++++++++++++++++++++++'''''+++++'''++++++++++++++##########################################+''''''''''''';+'''';;''''''''''''';';;;::::::::::;;;';;;;;;';;;;;;;;:::,,,,
++#######################+##+++++++++++++++++++++++++++'''''++++++''++++++++++++++#########################################++''''''''''''+':'+''';'''''''';'';'';;;:::::::::::;;;';;;;;;;;;;;;;;;:::,,,,

ChangelingRain reviewed (commented):
I think deep frying absolutely everything is terrible sorry

Iamgoofball committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into dab

# Conflicts:
#	_maps/map_files/MetaStation/MetaStation.dmm

GunHog commented:
Are you not going to make the fryer build-able?

Fox-McCloud commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,115 @@
+	create_reagents(50)
+	reagents.add_reagent("nutriment", 25)
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/machine/deep_fryer(null)
```
Correct me if I'm wrong, but isn't there a different way of doing this, now?

Iamgoofball replied:
IDK

Fox-McCloud commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,115 @@
+/obj/machinery/deepfryer/process()
+	..()
+	if (!reagents.has_reagent("nutriment"))
+		reagents.add_reagent("nutriment", 25)
```
And then the chef never bothered making anything other than deep-fried playing cards or metal rods for the crew.

This pretty much makes cooking anything pointless since you can just repeatedly shove insanely high availability items into the fryer and get a free 25 nutriment.

Iamgoofball replied:
And the vending machines didn't?

PKPenguin321 replied:
heres an idea to make this even more american
instead of nutrient make it apply "deep fat frier oil" which gives like a tenth of the hunger that nutrient does but makes you get fat faster (not sure if this is possible though since i think fatness is based on nutrition level but it'd be cool)

or just make it apply less than 25 nutrient who knows!

Armhulen replied:
Make deep frying turn it into junk food, problem solved.

Fox-McCloud commented:
deep frying everything is hilarious, but I reallyyy dislike how much it makes the chef's job pointless.

Basically it becomes a case of "get your hands on some high amount, high availability item" (like playing cards metal rods, or the like) and deep fry it to no end---now you have zero reason to make any legitimate food other than this.

Iamgoofball commented:
Vending machine already removed the point of making legit food.

Fox-McCloud commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,115 @@
+		if(user.drop_item() && !frying)
+			user << "<span class='notice'>You put [I] into [src].</span>"
+			frying = I
+			frying.loc = src
```
`frying.forceMove(src)`

Fox-McCloud commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,115 @@
+			S.overlays = frying.overlays
+			S.icon_state = frying.icon_state
+			S.desc = frying.desc
+			src.reagents.trans_to(S, 2*cook_time)
```
implied `src`

Fox-McCloud commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,115 @@
+			if(istype(frying, /obj/item/weapon/reagent_containers/food/snacks/))
+				qdel(frying)
+			else
+				frying.loc = S
```
`frying.forceMove(S)`

Also, why does it put the object being deepfried in the deepfried holder? It's never going to get interacted with; seems that it would be better to deleted.

The only reason I can think that this is a thing is if you could somehow deepfry mobs; then they could yell and scream while being unable to do anything about it.

Iamgoofball replied:
People didn't want it deleted outri

Fox-McCloud replied:
But...um, why? Aside from grenades, there's not much point to putting them in the food object...especially since that object is just going to get deleted when you finish eating the food item (it'll also mean that it could potentially cause food items to fail to GC because of this).

Iamgoofball replied:
It's what the people wanted. Not making the change.

Fox-McCloud commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -0,0 +1,115 @@
+
+			icon_state = "fryer_off"
+			user.put_in_hands(S)
+			S = null
```
Uhhh...

Haevacht commented:
Legit food fills you up faster and you can put healing shit in it. Fuck off, its still useful.

Fox-McCloud commented:
>Legit food fills you up faster and you can put healing shit in it. Fuck off, its still useful.

hypothetically yes, but people don't exactly go to the  chef for healing needs--they go to him for something to eat. 

This also disincentivizes the botanists, even more, from making things for the chef (as opposed to personal experimentation), as they have no reason to generate food items for the chef when...it's all within the chef's power to do so by himself ("why should I make him more when he can do it himself?")

Fox-McCloud commented:
@Iamgoofball 

Here's an idea. Why not just add a new reagent that's space soybean oil, that is extracted from soybeans---and the chef stats off with a jug of 50 units of it...it could provide less nutrition or something.

Deep frying would **require** said oil for you to be able to deepfry anything, but it would be a limited resource and require botany to supply you with additional resources for you to be able to keep it up.

Iamgoofball commented:
deep frying already requires and uses up reagents, and you have to keep restocking it with nutrient for it to actually turn into food. if you just dumped water in there it'd be pure water

Fox-McCloud commented:
>deep frying already requires and uses up reagents, and you have to keep restocking it with nutrient for it to actually turn into food. if you just dumped water in there it'd be pure water

https://github.com/tgstation/tgstation/pull/23618/files#diff-7fadd09507a5b8c6035f0b703afac718R56

It doesn't. It starts off with 25 nutriment, and if it ever runs out, it just adds 25 more to itself, out of thin air; so it doesn't require reagents, it just uses up reagents.

Iamgoofball commented:
Whoooops

GunHog commented:
So, is that a "No" on making them build-able?

Iamgoofball commented:
I'll make them buildable

DoctorPork commented:
I cant wait to make deepfried twinkie sandwiches.

optimumtact commented:
this code is awful, you have a lot of work to do

Iamgoofball commented:
@optimumtact there's only a few problems, can you please remove the 3 week delay tag?

Iamgoofball committed:
fixes

Iamgoofball commented:
>This will need a click cooldown.

I have added this but I can't reply to the comment for ??? reasons

Iamgoofball committed:
uiqwrhfhwfadkhfwrahbk

PKPenguin321 commented:
GODDD BLESS AMERRRRRICAAAAAA

Bawhoppen commented:
My belated response: He didn't put it in the table this time, Thank god.
